proj4/src/main/scala/geotrellis/proj4/CRS.scala

```diff
@@ -63,64 +170,56 @@ trait CRS extends Serializable {
-    * which identifies a CRS definition in the EPSG database.
-    * (Codes are read and handled as strings).
-    * </ul>
-    * An example of a valid CRS name is <tt>EPSG:3005</tt>.
-    * <p>
-    * @param name the name of a coordinate system, with optional authority prefix
-    * @return the [[CoordinateReferenceSystem]] corresponding to the given name
-    */
-  def fromName(name: String): CRS =
-    new CRS { val crs = crsFactory.createFromName(name) }
+   * Returns the WKT representation of the Coordinate Reference System
+   * @return
+   */
+  def toWKT(): String ={
+    epsgCode match {
+      case None=>"Unable to convert"
+      case Some(string) => fromNameToWKT(epsgCode.get)
+    }
+  }
 
   /**
-    * Creates a [[CoordinateReferenceSystem]]
-    * from a PROJ.4 projection parameter string.
-    * <p>
-    * An example of a valid PROJ.4 projection parameter string is:
-    * <pre>
-    * +proj=aea +lat_1=50 +lat_2=58.5 +lat_0=45 +lon_0=-126 +x_0=1000000 +y_0=0 +ellps=GRS80 +units=m
-    * </pre>
-    * @param proj4Params a PROJ.4 projection parameter string
-    * @return the specified [[CoordinateReferenceSystem]]
-    */
-  def fromString(proj4Params: String): CRS =
-    new CRS { val crs = crsFactory.createFromParameters(null, proj4Params) }
+   * Returns the WKT representation given an EPSG code in the format EPSG:[number]
+   * @param name
+   * @return
+   */
+  def fromNameToWKT(name :String): String ={
+    val code = name.split(":")(1)
+
+    val dataStream=Source.fromFile("proj4/src/main/resources/wkt/epsg.properties").getLines().toStream.dropWhile(line=>line.split("=")(0)!=code)
+
+    if(dataStream.length>0) {
+      val array = dataStream(0).split("=")
+      array(1)
+    }
+    else "Please enter a valid code"
+  }
 
   /**
-    * Creates a [[CoordinateReferenceSystem]]
-    * from a PROJ.4 projection parameter string.
-    * <p>
-    * An example of a valid PROJ.4 projection parameter string is:
-    * <pre>
-    * +proj=aea +lat_1=50 +lat_2=58.5 +lat_0=45 +lon_0=-126 +x_0=1000000 +y_0=0 +ellps=GRS80 +units=m
-    * </pre>
-    * @param name a name for this coordinate system.
-    * @param proj4Params a PROJ.4 projection parameter string
-    * @return the specified [[CoordinateReferenceSystem]]
-    */
-  def fromString(name: String, proj4Params: String): CRS =
-    new CRS{ val crs = crsFactory.createFromParameters(name, proj4Params) }
+   * Returns the numeric EPSG code of a proj4string
+   * @param proj4String
+   * @return
+   */
+  def getEPSGCodeOfProj4String(proj4String: String):Option[String]={
+    val filePrefix = "proj4/src/main/resources/nad/"
+
+    def code(line:String):Option[String]={
+      val array = line.split(" ")
+      val length = array(0).length
+      Some("EPSG:"+array(0).substring(1,length-1))
+    }
+
+    val dataStream =Source.fromFile(filePrefix+"epsg").getLines().toStream.dropWhile(line=>line.startsWith("#") || "+proj"+((line).split("proj"))(1)!=proj4String)
```

Comment:
Hi Rob,

If we do not keep the data in memory, anyway we'll have to sequentially check every line right ? 

But if we need these functions frequently then we can have them read only the first time and keep them in memory in some data structure like hash map. Furthermore we can keep several maps considering +proj value for a given string. 
When retrieving the string for a particular code, we can have the entries sorted at the beginning considering the code number and use binary search to retrieve the string.